iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -1093,5 +1100,16 @@ PortManager::acquireConditionVariableData(const RuntimeName_t& runtimeName) noex
     return m_portPool->addConditionVariableData(runtimeName);
 }
 
+bool PortManager::isInternal(const capro::ServiceDescription& service) const noexcept
+{
+    const capro::ServiceDescription serviceRegistry{
+        SERVICE_DISCOVERY_SERVICE_NAME, SERVICE_DISCOVERY_INSTANCE_NAME, SERVICE_DISCOVERY_EVENT_NAME};
+
+    return service == serviceRegistry || service == IntrospectionPortService || service == IntrospectionMempoolService
+           || service == IntrospectionMempoolService || service == IntrospectionPortThroughputService
+           || service == IntrospectionSubscriberPortChangingDataService;
+}
```

Comment:
Wouldn't it be much easier and future proof to reserve `iceoryx` as service ID for iceoryx and do not allow that anybody else uses this service ID?

Our services would then be renamed to
```
iceoryx - service-registry - all
iceoryy - introspection - SubscriberPortsData
iceoryx - introspection - PortThroughput
iceoryx - introspection - Process
iceoryx - introspection - MemPool
iceoryx - introspection - Port
```

This scales much better and we cannot forget to extend the deny-list. What do you think?

Reply:
That's definitely a possible solution, although a more restrictive one. However, I would propose to tackle this with either #518 or #1135 and not on this PR. If we do it before `v2.0` we might have to change the API again, because some things might not turn out as planned with the `StatusPort` (#1064). If we do it all together we just break the API once for the user. I'm talking from the POV of a `rmw_iceoryx` maintainer here ;)

Reply:
Could you add this a food for thought at one of the issues. Just to not forget about it.

Reply:
Already done ;)